### PR TITLE
[tests] Add test case for magnetometer sensor on FRDM-K64F

### PR DESCRIPTION
**Pull request by cuiyanx**
**TOTAL:** 11 of 11 passed

Signed-off-by: Cui Yan <yanx.cui@intel.com>

### CODE REVIEW

**Comment by grgustaf** [comment]
@jimmy-huang please review

File: modules/GenericSensor.js [hunk 1]
```diff
@@ -126,8 +127,13 @@ function GenericSensor() {
                            sensor.z !== null,
                            "sensor: reading value for '" + sensorType + "'");
 
-                    assert(sensor.controller === "bmi160",
-                           "sensor: controller name as default");
+                    if (sensorType === "Magnetometer") {
```
Comment (jimmy-huang) [inline modules/GenericSensor.js]: I think this was discussed in #1055, I don't think we should be checking the sensor.controller as it's not a property that will be part of the API.  The controller is only passed as an optional argument to the constructor, and once the object is created, it's no longer relevant.  @cuiyanx so can you maybe explain what is the need here to check the controller value after it has been set?  Maybe we need to update the API doc a little better to not confuse the developers.

Comment (cuiyanx) [inline modules/GenericSensor.js]: Li Hao and I have discussed this issue, so we support your point of view. I will delete this test point.

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**